website/src/documentation/runtime/environments.md

```diff
@@ -0,0 +1,187 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Runtime environments
+
+Any execution engine can run the Beam SDK beacuse the SDK runtime environment is [containerized](https://s.apache.org/beam-fn-api-container-contract) with [Docker](https://www.docker.com/) and isolated from other runtime systems. This page describes how to build, customize, and push Beam SDK container images.
+
+## Building container images
+
+Before building Beam SDK container images:
+* Register a [Bintray](https://bintray.com/) account with a Docker repository named `apache`.
+* Install [Docker](https://www.docker.com/) on your workstation.
+
+To build Beam SDK container images:
+
+<ol>
+    <li>
+        Navigate to your local copy of the <a href="https://github.com/apache/beam"><code>beam</code></a>
+    </li>
+    <li>
+        Run Gradle with the <code>docker</code> target: <pre>./gradlew docker</pre>
+    </li>
+</ol>
+
+> **Note**: It may take a long time to build all of the container images. You can instead build the images for specific SDKs:
+>
+> ```
+> ./gradlew -p sdks/java/container docker
+> ./gradlew -p sdks/python/container docker
+> ./gradlew -p sdks/go/container docker
+> ```
+
+Run `docker images` to examine the containers. For example, if you successfully built the container images, the command prompt displays a response like:
+
+```
+REPOSITORY                                       TAG                    IMAGE ID            CREATED       SIZE
+$USER-docker-apache.bintray.io/beam/python     latest             4ea515403a1a      3 minutes ago     1.27GB
+$USER-docker-apache.bintray.io/beam/java       latest             0103512f1d8f     34 minutes ago      780MB
+$USER-docker-apache.bintray.io/beam/go         latest             ce055985808a     35 minutes ago      121MB
+```
+
+<b>Although the respository names look like URLs, the container images are stored locally on your workstation.</b> After building the container images locally, you can [push](#pushing-container-images) them to an eponymous repository online.
+
+### Overriding default Docker targets
+
+The default SDK version is `latest` and the default Docker repository is the following Bintray location:
+
+```
+$USER-docker-apache.bintray.io/beam
+```
+
+When you [build SDK container images](#building-container-images), you can override the default version and location.
+
+To specify an older Python SDK version, like 2.3.0, build the container with the `docker-tag` option:
+
+```
+./gradlew docker -Pdocker-tag=2.3.0
+```
+
+To change the `docker` target, build the container with the `docker-repository-root` option:
+
+```
+./gradlew docker -Pdocker-repository-root=$LOCATION
+```
+
+## Customizing container images
+
+You can add extra dependencies or serialization files to container images so the execution engine doesn't need them.
```

Comment:
This sentence is unclear--

Do you mean that if you add extra dependencies or serialization files, then you don't need to supply them again later? What does it mean that "the execution engine doesn't need them"? What is "them" in this sentence?

Reply:
Updated to:

"You can add extra dependencies to container images so you don't have to supply the dependencies to execution engines."